### DISCUSSION TITLE
Recent Troubles with the Script (Formerly Working Great)

### DISCUSSION THREAD

**Discussion by SomeRandomUserIsHere** [jsaddiction/TrailerTech#50] [General]
Hello, I've been using your script for a year or so to much benefit, thank you!

However, I recently put a whole load of movies through Radarr and noticed I wasn't getting trailers. Some of it was my own fault (long story), but I've run into several other issues that I thought I'd mention here. Note I'm a newb at Python and Dockers, etc., so I'm way out of my element, forgive me if I butcher some of this.

1. Linux.io Docker images now treat the custom-cont-init.d thing differently I think. It requires loading that directory as a volume in the Docker itself (and I feel like I even had to move it out of /config). So, the auto install script doesn't work as described, it needs to be mounted as a volume. You can see in the Linux.io documentation.

2. I am using that install script on a Synology NAS, and it was working before, but this time Python wouldn't install, I could see in the logs. I had to change to this, and it seemed to work again:

OLD:   #install="git python3 python3-pip ffmpeg"
NEW:    install="git python3 py3-pip ffmpeg"

3. I started getting this error. Probably something I've done, but I'm too dumb to figure it out besides entering the exception. Any thoughts on what is going on here?

stdout | git config --global --add safe.directory /config/scripts/TrailerTech
stdout | To add an exception for this directory, call:
stdout | Failed to update local repo. ERROR: fatal: detected dubious ownership in repository at '/config/scripts/TrailerTech'

4. After all that, I thought I had it working again. Radarr "test" worked, but nothing. So ran from Command Line, and it looked like it was working, found a good YT URL, but then just hung on downloading. I tried pulling in a newer version of youtube-dl, but still having the problem. 

If you have any thoughts for helping, THANKS IN ADVANCE!  Your script has been awesome for me up until now, I can't believe no one has built something like a Trailarr, your your script has been the next best thing!

**Comment by jsaddiction** [comment]
First of all thank you for the kind words!  This project has been a labor of love for a while and use it myself.  I have encountered the same issues you have and have worked around them in the scripts.  I do not run this script on syno but the installation should be the same assuming you are using linuxserver.io version of Radarr.

1. My docker compose file maps /custom-cont-init.d and /config directories independently and not within each other exactly like you mentioned.
2. Linuxserver.io utilizes alpine linux now with their Radarr image.  Ensure you are using the latest version of my install script.  py3-pip is already accounted for.
3. This one is a little tricky.  I have commands in the script which should set the safe.directory config for git.  Works for me but your mileage may vary.  Again ensure you are using the latest version of the install script.  If it still doesn't work then you may need to run those commands manually.
`sudo docker exec <radarr container name> git config --global --add safe.directory /config/scripts/TrailerTech`
Then try to rebuild the container

Trailarr or a version of it is in the back of my mind.  I am primarily a python developer by hobby and investigating potential in a MERN stack powered version of this application which also handles all other possible metadata for both movies and tv shows (maybe music).  It would be a web app in docker which gives the user ways to select images, trailers and modify information for that media.

**Comment by SomeRandomUserIsHere** [comment]
Oh man, thank you!  That was just enough of a nudge to get it working, so pysched...

Again, I wish more people knew about this script because IMHO, it's by far the best trailer script out there for use with Radarr (or even stand alone).

MUCH APPRECIATED!

**Comment by SomeRandomUserIsHere** [comment]
For some reason TrailerTech broke for me upon recently upgrading my Linux.io Radarr docker container to the latest release. Interestingly, I can run TrailerTech no problem from the command line, but the custom script in Radarr is failing. Looks like possibly a permissions issue with youtube-dl (Radarr and the docker container itself runs under a DockerUser user/group, but when I run from command line, I've been using an admin user. Have you had any issues with the new version of Radarr or the Linux.io container? Thanks again for any help!

**Comment by jsaddiction** [comment]
Took some digging but I got it figured out please reference #51
